Hazelcast.Test/Hazelcast.Client.Test/NestedPredicateVersionedPortablesTest.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using Hazelcast.Client.Model;
+using Hazelcast.Config;
+using Hazelcast.Core;
+using Hazelcast.IO.Serialization;
+using Hazelcast.Util;
+using NUnit.Framework;
+
+namespace Hazelcast.Client.Test
+{
+    [TestFixture]
+    [Category("3.8")]
+    public class NestedPredicateVersionedPortablesTest : SingleMemberBaseTest
+    {
+        private static IMap<int, Body> map;
+
+        [SetUp]
+        public void Init()
+        {
+            map = Client.GetMap<int, Body>("map");
+        }
+
+        [TearDown]
+        public static void Destroy()
+        {
+            map.Clear();
+        }
+
+        protected override void ConfigureClient(ClientConfig config)
+        {
+            base.ConfigureClient(config);
+            config.GetSerializationConfig().AddPortableFactory(1, new SimplePortableFactory());
+        }
+
+        private class SimplePortableFactory : IPortableFactory
+        {
+            public IPortable Create(int classId)
+            {
+                switch (classId)
+                {
+                    case 1:
+                        return new Body();
+                    case 2:
+                        return new Limb();
+                    default:
+                        throw new InvalidOperationException("Wrong class ID");
+                }
+            }
+        }
+
+
+        [Test]
+        public void addingIndexes()
+        {
+            // single-attribute index
+            map.AddIndex("name", true);
+            // nested-attribute index
+            map.AddIndex("limb.name", true);
+        }
+        
+        /* TODO
+        [Test]
+        public void singleAttributeQuery_versionedProtables_predicates()
```

Comment:
Are we missing a test or is it left over code?

Reply:
In this test, EntryObject and PredicateBuilder classes are used but there are no equivalent classes in C#. This test should be added when related classes are implemented.

Actually, I was not sure about what should I do, then I left it as commented out for your reviews.

Reply:
You can use predicate as :
```
var predicate = Predicates.Property("limb.name").Equal("hand");
```